components/camel-ssh/src/test/java/org/apache/camel/component/ssh/SshComponentKnownHostTest.java

```diff
@@ -0,0 +1,124 @@
+        mock.expectedBodiesReceived(msg);
+        mock.expectedHeaderReceived(SshResult.EXIT_VALUE, 0);
+        mock.expectedHeaderReceived(SshResult.STDERR, "Error:test\n");
+
+        template.sendBody("direct:ssh", msg);
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testProducerWithInvalidFile() throws Exception {
+        final String msg = "test\n";
+
+        MockEndpoint mock = getMockEndpoint("mock:password");
+        mock.expectedMessageCount(0);
+
+        template.sendBody("direct:sshInvalid", msg);
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testProducerWithInvalidFileWarnOnly() throws Exception {
+        final String msg = "test\n";
+
+        MockEndpoint mock = getMockEndpoint("mock:password");
+        mock.expectedMinimumMessageCount(1);
+        mock.expectedBodiesReceived(msg);
+        mock.expectedHeaderReceived(SshResult.EXIT_VALUE, 0);
+        mock.expectedHeaderReceived(SshResult.STDERR, "Error:test\n");
+
+        template.sendBody("direct:sshInvalidWarnOnly", msg);
+
+        assertMockEndpointsSatisfied();
+    }
+    
+    @Test
+    public void testPollingConsumerWithValidKnownHostFile() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:result");
+        mock.expectedMinimumMessageCount(1);
+        mock.expectedBodiesReceived("test\r");
+        mock.expectedHeaderReceived(SshResult.EXIT_VALUE, 0);
+        mock.expectedHeaderReceived(SshResult.STDERR, "Error:test\r");
+        assertMockEndpointsSatisfied();
+    }
+    
+    @Test
+    public void testPollingConsumerWithInvalidKnownHostFile() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:resultInvalid");
+        mock.expectedMessageCount(0);
+        assertMockEndpointsSatisfied();
+    }
+    
+    @Test
+    public void testPollingConsumerWithInvalidKnownHostFileWarnOnly() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:resultInvalidWarnOnly");
+        mock.expectedMinimumMessageCount(1);
+        mock.expectedBodiesReceived("test\r");
+        mock.expectedHeaderReceived(SshResult.EXIT_VALUE, 0);
+        mock.expectedHeaderReceived(SshResult.STDERR, "Error:test\r");
+        assertMockEndpointsSatisfied();
+    }
+
+    @Override
+    protected RouteBuilder createRouteBuilder() throws Exception {
+        return new RouteBuilder() {
+            @Override
+            public void configure() {
+                onException(Exception.class).handled(true).to("mock:error");
+
+                from("ssh://smx:smx@localhost:" + port + "?useFixedDelay=true&delay=40000&pollCommand=test%0D&knownHostsResource=classpath:known_hosts_valid&failOnUnknownHost=true")
+                    .to("mock:result");
+                
+                from("ssh://smx:smx@localhost:" + port + "?useFixedDelay=true&delay=40000&pollCommand=test%0D&knownHostsResource=classpath:known_hosts_invalid&failOnUnknownHost=true")
```

Comment:
I don't see a problem with that, since it uses the well-tested ResourceHelper class from camel-core. According to the javadoc it supports file: and http: URIs